### PR TITLE
Publicize isSlot<T>, isSignal<T>, isSlotGetter<T>, and isSignalGetter<T> (#675)

### PR DESCRIPTION
**Pull request by dalboris**
#675

### CODE REVIEW

**Comment by dalboris** [comment]
I also added a to-do note to change the type name `SlotRef` to `Slot` and `SignalRef` to `Signal`.

The idea is that the terminology "slot" (resp "signal") would mean the combination of a pointer to an Object's method together with a pointer to an Object instance, so that we can call it. 

This is consistent with the existing `SlotId`:

```
using FunctionId = Int;
using ObjectSlotId = std::pair<Object*, FunctionId>;
using FreeFuncId = void*;
using SlotId = std::variant<std::monostate, ObjectSlotId, FreeFuncId>;
```

(if we want to be pedantic, it could be called `ObjectSlot` rather than just `Slot`, if we want a free function to be included in the definition of "what is a slot", but better keep it simple)

In other words, the returned value of the slot getter is more than a "reference to the slot", it is the actual slot, that a signal can be connected to (the slot is callable with the arguments of the signal).

If I remember correctly, we initially called it `SlotRef` to avoid confusion with the method that returns the `SlotRef`: it was the method that we called the "slot". But with hindsight, we can instead call the method "slot getter" and its returns value the "slot".
